pandas/tests/indexes/multi/test_operations.py

```diff
@@ -0,0 +1,448 @@
+    # fill_value
+    result = idx.take(np.array([1, 0, -1]), fill_value=True)
+    exp_vals = [('A', pd.Timestamp('2011-01-02')),
+                ('A', pd.Timestamp('2011-01-01')),
+                (np.nan, pd.NaT)]
+    expected = pd.MultiIndex.from_tuples(exp_vals, names=['str', 'dt'])
+    tm.assert_index_equal(result, expected)
+
+    # allow_fill=False
+    result = idx.take(np.array([1, 0, -1]), allow_fill=False,
+                      fill_value=True)
+    exp_vals = [('A', pd.Timestamp('2011-01-02')),
+                ('A', pd.Timestamp('2011-01-01')),
+                ('B', pd.Timestamp('2011-01-02'))]
+    expected = pd.MultiIndex.from_tuples(exp_vals, names=['str', 'dt'])
+    tm.assert_index_equal(result, expected)
+
+    msg = ('When allow_fill=True and fill_value is not None, '
+           'all indices must be >= -1')
+    with tm.assert_raises_regex(ValueError, msg):
+        idx.take(np.array([1, 0, -2]), fill_value=True)
+    with tm.assert_raises_regex(ValueError, msg):
+        idx.take(np.array([1, 0, -5]), fill_value=True)
+
+    with pytest.raises(IndexError):
+        idx.take(np.array([1, -5]))
+
+
+def test_iter(idx):
+    result = list(idx)
+    expected = [('foo', 'one'), ('foo', 'two'), ('bar', 'one'),
+                ('baz', 'two'), ('qux', 'one'), ('qux', 'two')]
+    assert result == expected
+
+
+def test_sub(idx):
+
+    first = idx
+
+    # - now raises (previously was set op difference)
+    with pytest.raises(TypeError):
+        first - idx[-3:]
+    with pytest.raises(TypeError):
+        idx[-3:] - first
+    with pytest.raises(TypeError):
+        idx[-3:] - first.tolist()
+    with pytest.raises(TypeError):
+        first.tolist() - idx[-3:]
+
+
+def test_argsort(idx):
+    result = idx.argsort()
+    expected = idx.values.argsort()
+    tm.assert_numpy_array_equal(result, expected)
+
+
+def test_map(idx):
+    # callable
+    index = idx
+
+    # we don't infer UInt64
+    if isinstance(index, pd.UInt64Index):
+        expected = index.astype('int64')
+    else:
+        expected = index
+
+    result = index.map(lambda x: x)
+    tm.assert_index_equal(result, expected)
+
+
+@pytest.mark.parametrize(
+    "mapper",
+    [
+        lambda values, idx: {i: e for e, i in zip(values, idx)},
+        lambda values, idx: pd.Series(values, idx)])
+def test_map_dictlike(idx, mapper):
+
+    if isinstance(idx, (pd.CategoricalIndex, pd.IntervalIndex)):
+        pytest.skip("skipping tests for {}".format(type(idx)))
+
+    identity = mapper(idx.values, idx)
+
+    # we don't infer to UInt64 for a dict
+    if isinstance(idx, pd.UInt64Index) and isinstance(identity, dict):
+        expected = idx.astype('int64')
+    else:
+        expected = idx
+
+    result = idx.map(identity)
+    tm.assert_index_equal(result, expected)
+
+    # empty mappable
+    expected = pd.Index([np.nan] * len(idx))
+    result = idx.map(mapper(expected, idx))
+    tm.assert_index_equal(result, expected)
+
+
+def test_numpy_ufuncs(idx):
```

Comment:
I would just combine test_operations and test_analytics (call it test_analytics)